examples/text_prediction/run_text_prediction.py

```diff
@@ -46,49 +54,90 @@ def get_parser():
     else:
         raise NotImplementedError
     if args.exp_dir is None:
-        args.exp_dir = 'autogluon_{}'.format(args.task)
-    model = task.fit(train_data=args.train_file,
-                     label=label_columns,
-                     feature_columns=feature_columns,
-                     output_directory=args.exp_dir,
-                     stopping_metric=stop_metric,
-                     ngpus_per_trial=1,
-                     eval_metric=eval_metrics)
-    dev_metrics_scores = model.evaluate(args.dev_file, metrics=eval_metrics)
-    with open(os.path.join(args.exp_dir, 'final_model_dev_score.json'), 'w') as of:
-        json.dump(dev_metrics_scores, of)
-    dev_prediction = model.predict(args.dev_file)
-    with open(os.path.join(args.exp_dir, 'dev_predictions.txt'), 'w') as of:
-        for ele in dev_prediction:
-            of.write(str(ele) + '\n')
-    model.save(os.path.join(args.exp_dir, 'saved_model'))
-    model = task.load(os.path.join(args.exp_dir, 'saved_model'))
-    test_prediction = model.predict(args.test_file)
-    with open(os.path.join(args.exp_dir, 'test_predictions.txt'), 'w') as of:
-        for ele in test_prediction:
-            of.write(str(ele) + '\n')
+        args.exp_dir = 'autogluon_text_{}'.format(args.task)
+    train_df = load_pd.load(args.train_file)
+    dev_df = load_pd.load(args.dev_file)
+    test_df = load_pd.load(args.test_file)
+    train_df = train_df[feature_columns + [label_column]]
+    dev_df = dev_df[feature_columns + [label_column]]
+    test_df = test_df[feature_columns]
+    if args.task == 'mrpc' or args.task == 'sts':
+        # Augmenting the un-ordered set manually.
+        train_df_other_part = pd.DataFrame({feature_columns[0]: train_df[feature_columns[1]],
+                                            feature_columns[1]: train_df[feature_columns[0]],
+                                            label_column: train_df[label_column]})
+        real_train_df = pd.concat([train_df, train_df_other_part])
+        real_dev_df = dev_df
+    else:
+        real_train_df = train_df
+        real_dev_df = dev_df
+    if args.mode == 'stacking':
+        predictor = TabularPredictor(label=label_column,
+                                     eval_metric=eval_metric,
+                                     path=args.exp_dir)
+        predictor.fit(train_data=real_train_df,
+                      tuning_data=real_dev_df,
+                      hyperparameters='multimodal',
+                      num_bag_folds=5,
+                      num_stack_levels=1)
+    elif args.mode == 'weighted':
+        predictor = TabularPredictor(label=label_column,
+                                     eval_metric=eval_metric,
+                                     path=args.exp_dir)
+        predictor.fit(train_data=real_train_df,
+                      tuning_data=real_dev_df,
+                      hyperparameters='multimodal')
+    elif args.mode == 'single':
```

Comment:
May want to add comment that highlights why TextPredictor is used for 'single', TabularPredictor for 'weighted', 'stacking'. Otherwise this is easy to miss.

Reply:
added the comment.